pkg/cli/admin/prune/images/images.go

```diff
@@ -436,7 +436,14 @@ func (o PruneImagesOptions) Run() error {
 		if len(registryHost) == 0 {
 			registryHost, err = imageprune.DetermineRegistryHost(allImages, allStreams)
 			if err != nil {
-				return fmt.Errorf("unable to determine registry: %v", err)
+				if err != imageprune.ErrNoImageFound {
+					return fmt.Errorf("unable to determine registry: %v", err)
+				}
+				// if we can't determine the registry host (ErrNoImageFound) and
+				// we need to prune it as well, bail out immediately.
+				if o.PruneRegistry != nil && *o.PruneRegistry {
+					return fmt.Errorf("unable to find the remote registry host: %v", err)
+				}
```

Comment:
@ricardomaraschini how will this prevent the cluster from being degraded? Will `o.PruneRegistry` be `false` and/or nil if the image registry is `Removed`?

Reply:
The problem here is that even with `--prune-registry=false` it was still attempting and failing to parse the registry URL.